runtime/cudaq/platform/default/rest_server/helpers/RestRemoteServer.cpp

```diff
@@ -93,21 +95,88 @@ class RemoteRestRuntimeServer : public cudaq::RemoteRuntimeServer {
       throw std::runtime_error(
           "Invalid TCP/IP port requested. Valid range: [1024, 65535].");
     m_server = std::make_unique<cudaq::RestServer>(m_port);
-    m_server->addRoute(cudaq::RestServer::Method::GET, "/",
-                       [](const std::string &reqBody) {
-                         // Return an empty JSON string,
-                         // e.g., for client to ping the server.
-                         return json();
-                       });
+    m_server->addRoute(
+        cudaq::RestServer::Method::GET, "/",
+        [](const std::string &reqBody,
+           const std::unordered_multimap<std::string, std::string> &headers) {
+          // Return an empty JSON string,
+          // e.g., for client to ping the server.
+          return json();
+        });
 
     // New simulation request.
-    m_server->addRoute(cudaq::RestServer::Method::POST, "/job",
-                       [&](const std::string &reqBody) {
-                         std::string mutableReq = reqBody;
-                         if (m_hasMpi)
-                           cudaq::mpi::broadcast(mutableReq, 0);
-                         return processRequest(reqBody);
-                       });
+    m_server->addRoute(
+        cudaq::RestServer::Method::POST, "/job",
+        [&](const std::string &reqBody,
+            const std::unordered_multimap<std::string, std::string> &headers) {
+          std::string mutableReq;
+          for (const auto &[k, v] : headers)
+            cudaq::info("Request Header: {} : {}", k, v);
+          // Checking if this request has its body sent on as NVCF assets.
+          const auto dirIter = headers.find("NVCF-ASSET-DIR");
+          const auto assetIdIter = headers.find("NVCF-FUNCTION-ASSET-IDS");
+          if (dirIter != headers.end() && assetIdIter != headers.end()) {
+            const std::string dir = dirIter->second;
+            const auto ids = cudaq::split(assetIdIter->second, ',');
+            if (ids.size() != 1) {
+              json js;
+              js["status"] =
+                  fmt::format("Invalid asset Id data: {}", assetIdIter->second);
+              return js;
+            }
```

Comment:
This is just a sanity check for the `NVCF-FUNCTION-ASSET-IDS` field against edge cases like wrong requests with multiple assets hitting our function. It'd fail anyway since that wouldn't be a valid file name but it'd be nice to catch it early here.